api/post.go

```diff
@@ -129,7 +131,7 @@ func CreateValetPost(c *Context, post *model.Post) (*model.Post, *model.AppError
 	return rpost, nil
 }
 
-func CreatePost(c *Context, post *model.Post, doUpdateLastViewed bool) (*model.Post, *model.AppError) {
+func CreatePost(c *Context, post *model.Post, teamId string, doUpdateLastViewed bool) (*model.Post, *model.AppError) {
```

Comment:
Why are we passing teamId and Context? TeamId is in the context. `c.Session.TeamId`